Riot/Modules/Application/AppCoordinator.swift

```diff
@@ -259,6 +263,35 @@ final class AppCoordinator: NSObject, AppCoordinatorType {
         // Reload split view with selected space id
         self.splitViewCoordinator?.start(with: spaceId)
     }
+    
+    private func setupPushRulesSync() {
+        let sessionReady = NotificationCenter.default.publisher(for: .mxSessionStateDidChange)
+            .compactMap { $0.object as? MXSession }
+            .filter { $0.state == .running }
+            .removeDuplicates { session1, session2 in
+                session1 == session2
+            }
+        
+        sessionReady
+            .sink { [weak self] session in
+                let applicationDidBecomeActive = NotificationCenter.default.publisher(for: UIApplication.didBecomeActiveNotification).eraseOutput()
+                let needsCheckPublisher = applicationDidBecomeActive.merge(with: Just(())).eraseToAnyPublisher()
+                
+                self?.pushRulesUpdater = .init(notificationSettingsService: MXNotificationSettingsService(session: session), needsCheck: needsCheckPublisher)
```

Comment:
Wouldn't it be simpler to sink `didBecomeActiveNotification` in here and just call `syncRulesIfNeeded` on the `pushRulesUpdater` without faffing around with publishers? 🤔

Reply:
Yeah, I was already thinking on that before you wrote this comment actually. 
Applied -> https://github.com/vector-im/element-ios/pull/7335/commits/3972da892e9dd6362dd70990e408bea7afaca419